lib/elixir_sense/core/binding.ex

```diff
@@ -154,6 +154,34 @@ defmodule ElixirSense.Core.Binding do
     end
   end
 
+  # dependency injection
+  def do_expand(env, {:call, {:atom, Application}, fun, args}, stack)
+      when fun in ~w(get_env get_env! fetch_env fetch_env!)a do
+    try do
+      mod = apply(Application, fun, Enum.map(args, &elem(&1, 1)))
+
+      case mod do
+        {:ok, mod} ->
+          expand(env, {:atom, mod}, stack)
```

Comment:
you can skip expand here and return, BTW return type is {:ok, atom} and not atom
```suggestion
          {:tuple, 2, {:atom, :ok}, {:atom, atom}}
```

Reply:
I reflected a bit on this suggestion. My first impression was that it should still be `{:atom, atom}` and not `{:tuple, 2, {:atom, :ok}, {:atom, atom}}`, but after reflecting, it should be a tuple. But if it's a tuple, the module attribute can't be used directly as a dependency, so there shouldn't be dependency injection with `fetch_env/2`—which returns the `{:ok, mod}` tuple.

So I'm thinking of removing the inference for module attributes that uses `fetch_env/2` and only inferring when using `get_env!/2`, `get_env/3` and `fetch_env!/2`. What do you think?

Reply:
It doesn't make the code much more complex so I'd leave it just for completeness sake.

Reply:
I don't see the benefits of having this here. I can't think of a case where someone would inject a dependency using `fetch_env/2`, and this code is related to dependency injection inference.

If there's a benefit that is not related to dependency injection inference, I believe it would make more sense to do that in another PR. Maybe adding a new `do_expand/3` case matching exactly to `fetch_env/2`.

Is there a benefit related to dependency injection that I may be missing?